### PR TITLE
Drop shardalias

### LINKED ISSUE #746: Drop shardalias column from pg_dist_shard
It's been deprecated and is no longer used by the code. Once that change is incorporated into a release (5.3), the next release (5.4) should drop the `shardalias` column.

### PR DESCRIPTION
**Pull request by lithp**
Fixes #746

### CODE REVIEW

**Comment by marcocitus** [comment]
There shardalias fields pg_dist_partition.h. My understanding is that we need to keep those because we will always have the dropped column in the catalog, is that correct? (If so, might be worth clarifying in a comment)

**Comment by lithp** [comment]
Good point, how do you like this?

File: src/backend/distributed/master/master_metadata_utility.c [hunk 1]
```diff
@@ -379,8 +379,8 @@ InsertShardRow(Oid relationId, uint64 shardId, char storageType,
 	values[Anum_pg_dist_shard_shardid - 1] = Int64GetDatum(shardId);
 	values[Anum_pg_dist_shard_shardstorage - 1] = CharGetDatum(storageType);
 
-	/* deprecated shardalias column is always null. */
-	isNulls[Anum_pg_dist_shard_shardalias_DEPRECATED - 1] = true;
+	/* dropped shardalias column must also be set or pg will crash */
```
Comment (marcocitus) [inline src/backend/distributed/master/master_metadata_utility.c]: Maybe: or pg will crash -> Because it is still part of the tuple descriptor.

**Comment by marcocitus** [comment]
:ship:

**Pull request merged**